src/main/webapp/components/sources-info/sources-info.js

```diff
@@ -44,41 +55,77 @@ const sourcesMessage = offlineCount => {
 
   return `${offlineCount} sources are currently down`
 }
-
 export const SourcesInfo = props => {
   const sources = props.sources === undefined ? [] : props.sources
-  const offlineCount = sources.filter(source => !source.isAvailable).length
-  const getIcon = source => (source.isAvailable ? OnlineIcon : OfflineIcon)
-  if (props.loading) {
+  const SourceSet = new Set(sources);
+  const getIcon = source => {
+    if(source.__typename == 'FacetResult'){
+      return StorageIcon
+    }
+    if(source.isAvailable){
+      return  OnlineIcon 
+    }
+    else{  
+      return OfflineIcon
+    }
+  }
+  const offlineCount = Array.from(SourceSet).filter(source => !source.isAvailable).length
+  
+  const { data, loading, error, refetch } = useQuery(facetedQuery, {
+    variables: { attribute: FACET_ATTRIBUTE },
+  })
```

Comment:
We typically put GraphQL requests in the container to keep separation between logic and presentation. i.e. in here for this component: https://github.com/connexta/revelio/blob/master/src/main/webapp/components/sources-info/sources-info.js#L97